x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1202,7 +1203,13 @@ Collection<Object> createComponents(
         return components;
     }
 
-    private void applyPendingSecurityMigrations(SecurityIndexManager.State newState) {
+    private void applyPendingSecurityMigrations(SecurityIndexManager.State oldState, SecurityIndexManager.State newState) {
+        if (oldState.creationTime == null) {
+            // Bypass migrations for when the security index is new
+            submitPersistentMigrationTask(SecurityMigrations.MIGRATIONS_BY_VERSION.lastKey(), false);
+            return;
```

Comment:
When the index is new, we tell the migration executor to not apply any of the migrations, just update cluster state with the latest version since there is no data to migrate.

Reply:
What if the .security index does not exist AND is also not created soon? What would this update do in that case?

Reply:
Why cannot we let a regular migration be a noop when the index is just created and instead we create this new code path that we have to test and maintain?